sdk/rpc/server/server.go

```diff
@@ -136,17 +137,18 @@ func (t *RPCServer) CredentialProvisionerDescription(req proto.ProvisionerID, re
 // interface. The call is forwarded to the Provision() function of the Provisioner of the credential identified by
 // req.CredentialID.
 func (t *RPCServer) CredentialProvisionerProvision(req proto.ProvisionCredentialRequest, resp *sdk.ProvisionOutput) error {
+	defer func() {
+		if err := recover(); err != nil {
+			resp.AddError(fmt.Errorf("your locally built plugin failed with the following panic: %s; and with stack trace: %s", err, string(debug.Stack())))
+		}
+	}()
 	provisioner, err := t.getProvisioner(req.ProvisionerID)
 	if err != nil {
 		return err
 	}
-	*resp = sdk.ProvisionOutput{
-		Environment: make(map[string]string),
-		CommandLine: nil,
-		Files:       make(map[string]sdk.OutputFile),
-		Diagnostics: sdk.Diagnostics{},
-	}
+	*resp = req.ProvisionOutput
```

Comment:
This is a breaking change as it sets a strict requirement on the `req.ProvisionOutput` field. 

For consideration: if we move the initialization to a shared function instead, we can achieve the same without having to break the API. Alternatively, we have to bump `proto.Version`.

Reply:
If we are changing the API, let's make sure to change it for all requests.

Reply:
Question here (just by looking at this): even if we move the initialisation to a shared function, we would still need to set a strict requirement for `req.CommandLine` field ([see Flo's adjustment](https://github.com/1Password/shell-plugins/commit/6bc2bd73ca63c50288c93c31ea9dfb2dda0ccd5e)) in order to pass the command line arguments, is this correct?

If this is the case, I think this prompts for bumping the `proto.Version` either way.

Reply:
That's fair. I am not 100% happy with the fact that `req.CommandLine` (an output) is populated at the start of `Provision()`, but we can tackle that later and separately. 

The most important thing for now is that we make sure to merge this into the `protocol-v3` branch.

Reply:
This is indeed a protocol breaking change. I agree with Horia that there is no way around incrementing `port.Version`. The most elegant solution here (which would also avoid breaking protocol) would have been to use the `resp` parameter of the function directly as parameter for the `Provision` call here, but that pointer ends up pointing to a new struct when the rpc call is made. That is why I had to pass the output as a struct through the request structure.